### PR TITLE
Add `pihole-cloudflared` && `pihole-unbound`

### PR DESCRIPTION
**Pull request by aelfa**

### CODE REVIEW

File: apps/selfhosted/pihole-cloudflared.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - proxy
+    dns:
+      - ${NS1}
+      - ${NS2}
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'PIHOLE_DNS_=172.18.0.200#5053;172.18.0.200#5053'
```
Comment (doob187) [inline apps/selfhosted/pihole-cloudflared.yml:26]: DNS_ ?

**Review by doob187 — CHANGES_REQUESTED** [review]
See  note

File: apps/selfhosted/pihole-cloudflared.yml (hunk 1 above)
Comment (aelfa) [reply apps/selfhosted/pihole-cloudflared.yml:26]: @doob187 This uses either `cloudflared_DOH` or `unbound` as the upstream DNS server over the internal network.

Comment (aelfa) [reply apps/selfhosted/pihole-cloudflared.yml:26]: Example if you wanted to use simple Cloudflare & google here>
```yaml
      - 'PIHOLE_DNS_=1.1.1.1;8.8.8.8'
```

Comment (aelfa) [reply apps/selfhosted/pihole-cloudflared.yml:26]: @doob187 https://github.com/pi-hole/docker-pi-hole/#upgrade-notes

**Review by doob187 — APPROVED** [review]

**Pull request merged**